src/components/addSourceWizard/hardcodedComponents/aws/provisioningArn.js

```diff
@@ -0,0 +1,226 @@
+      "ec2:CreateKeyPair",
+      "ec2:CreateLaunchTemplate",
+      "ec2:CreateLaunchTemplateVersion",
+      "ec2:CreateTags",
+      "ec2:DeleteKeyPair",
+      "ec2:DeleteTags",
+      "ec2:DescribeAvailabilityZones",
+      "ec2:DescribeImages",
+      "ec2:DescribeInstanceTypes",
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeLaunchTemplates",
+      "ec2:DescribeLaunchTemplateVersions",
+      "ec2:DescribeRegions",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeSnapshotAttribute",
+      "ec2:DescribeTags",
+      "ec2:ImportKeyPair",
+      "ec2:RunInstances",
+      "ec2:StartInstances"
+    ],
+    "Resource": "*"
+  }
+]
+}' | jq '.Policy.Arn')`;
+
+const ROLE_COMMAND_STRING = `
+ROLE_ARN=$(aws iam create-role --role-name ${ROLE_NAME} --assume-role-policy-document '{
+  "Version": "2012-10-17",
+  "Statement": {
+    "Effect": "Allow",
+    "Principal": {
+      "AWS": "<PROVISIONING_AWS_ACCOUNT>"
+    },
+    "Action": "sts:AssumeRole"
+  }
+}' | jq '.Role.Arn')`;
+
+const ATTACH_COMMAND_STRING = `aws iam attach-role-policy --role-name ${ROLE_NAME} --policy-arn $POLICY_ARN`;
+
+export const IAMPolicyDescription = () => {
+  const intl = useIntl();
+
+  return (
+    <TextContent>
+      <Text component={TextVariants.p}>
+        {intl.formatMessage({
+          id: 'provisioning.iam.grantPermissions',
+          defaultMessage:
+            'To grant Red Hat permissions to RunInstances on your Amazon Web Services (AWS) Elastic Cloud (EC2), create following AWS Identity and Access Management (IAM) policy.',
+        })}
+      </Text>
+      <TextList component={TextListVariants.ol}>
+        <TextListItem>
+          {intl.formatMessage(
+            {
+              id: 'provisioning.iam.signIn',
+              defaultMessage: 'Log in to AWS CLI by running: {awsCliCommand}.',
+            },
+            {
+              awsCliCommand: (
+                <b>
+                  {intl.formatMessage({
+                    id: 'provisioning.iam.signInCommand',
+                    defaultMessage: 'aws configure',
+                  })}
+                </b>
+              ),
+            }
+          )}
+        </TextListItem>
+        <TextListItem>
+          {intl.formatMessage({
+            id: 'provisioning.iam.createPolicty',
+            defaultMessage: 'Create a new policy and store its ARN by running following command.',
+          })}
+        </TextListItem>
+      </TextList>
+      <ClipboardCopy isCode variant={ClipboardCopyVariant.expansion} className="pf-u-m-sm-on-sm" isReadOnly>
+        {POLICY_COMMAND_STRING}
+      </ClipboardCopy>
+    </TextContent>
+  );
+};
+
+export const IAMRoleDescription = () => {
+  const intl = useIntl();
+
+  const [provAppTypeId, setAppTypeId] = useState();
+  const [provAwsAccount, setAwsAccount] = useState(null);
+  const [fetchError, setFetchError] = useState(null);
+
+  useEffect(() => {
+    getSourcesApi()
+      .getProvAppType()
```

Comment:
This call could possibly be saved if we could get the data for `applicationTypes` that are already fetched in the `schemaBuilder`, but I'm not sure how to get to those data from this component.

Reply:
I've opened a PR #1032 to move to a react-query, with which I'd be able to reuse the data for applicationTypes and filter my app on the client.

Reply:
I've added a comment to the query PR. There is currently a redux in place for global state management. That would be the place to store/get the data from.